### PR TITLE
WIP: community/php7: upgrade to 7.3.6

### PR DESCRIPTION
**Pull request by bratkartoffel**
This PR updates PHP to the latest minor version (7.3.6).

A summary of the changes can be found at the php website: https://secure.php.net/manual/en/migration73.php

Changelogs:
https://secure.php.net/ChangeLog-7.php#7.3.0
https://secure.php.net/ChangeLog-7.php#7.3.1
https://secure.php.net/ChangeLog-7.php#7.3.2
https://secure.php.net/ChangeLog-7.php#7.3.3
https://secure.php.net/ChangeLog-7.php#7.3.4
https://secure.php.net/ChangeLog-7.php#7.3.5
https://secure.php.net/ChangeLog-7.php#7.3.6

### CODE REVIEW

**Comment by andypost** [comment]
This will make sense after 3.9 some packages depends on php and can't work with 7.3

The goal for 3.9 release is https://github.com/alpinelinux/aports/pull/5569

**Comment by andypost** [comment]
Another 7.3 extensions checked:
- php7-pecl-apcu 
- php7-pecl-msgpack
- php7-pecl-lzf

File: community/php7/APKBUILD [hunk 1]
```diff
@@ -75,6 +75,7 @@ makedepends="
 	net-snmp-dev
 	openldap-dev
 	pcre-dev
```
Comment (andypost) [inline community/php7/APKBUILD]: are you sure first version still needed?

Comment (bratkartoffel) [reply community/php7/APKBUILD]: I haven't checked all the dependencies yet, but will try to minimize them

**Comment by bratkartoffel** [comment]
php7-pecl-apcu: Needs patch
php7-pecl-msgpack: OK
php7-pecl-lzf: OK

**Comment by andypost** [comment]
`php7-pecl-xdebug` https://bugs.xdebug.org/view.php?id=1583

**Comment by andypost** [comment]
One more app 7.3 compatible https://github.com/alpinelinux/aports/pull/6684

**Comment by andypost** [comment]
Package `php7-pear-auth_sasl` could be removed as outdated https://pear.php.net/package/Auth_SASL @rnalrd

**Comment by andypost** [comment]
`community/php7-pear-net_socket` is not compatible with 7.3 https://github.com/alpinelinux/aports/pull/6708

**Comment by andypost** [comment]
`community/php7-pear-mail_mime` also need testing https://github.com/alpinelinux/aports/pull/6710

**Comment by andypost** [comment]
php-ssh2 should be ready for php 7.3 https://github.com/alpinelinux/aports/pull/6226

**Comment by TBK** [comment]
Would be great if this could be merged before 3.10.

**Comment by TBK** [comment]
btw 7.3.4 is out https://secure.php.net/ChangeLog-7.php#7.3.4

**Comment by bratkartoffel** [comment]
@TBK thanks for taking a look, I've upgraded the PR to 7.3.4
Yes, you're right, we should start merging this and the other PRs:

https://github.com/alpinelinux/aports/pull/5901
https://github.com/alpinelinux/aports/pull/5902
https://github.com/alpinelinux/aports/pull/5905

**Comment by TBK** [comment]
7.3.5 is out https://secure.php.net/ChangeLog-7.php#7.3.5

**Comment by bratkartoffel** [comment]
@TBK Upgraded to 7.3.5 and rebased, thanks

**Comment by TBK** [comment]
Do we want all the static libs? https://pkgs.alpinelinux.org/contents?file=*.a&path=&name=php7-dev&branch=edge&arch=x86_64

If then I suggest we add a $pkgname-static subpkg.

**Comment by TBK** [comment]
Did not know of PR https://github.com/alpinelinux/aports/pull/7809 before writing my last comment.

**Comment by TBK** [comment]
3.7.6 is out https://secure.php.net/ChangeLog-7.php#7.3.6

**Comment by Ikke** [comment]
@bratkartoffel Is this still WIP?

**Comment by andypost** [comment]
@Ikke yes, waiting for 3.10 release

**Comment by captn3m0** [comment]
Ignore if I'm misunderstanding this, but does this mean Alpine 3.10 will ship with PHP 7.2?

[7.2 Active Support is ending in December this year](https://endoflife.date/php) and 7.2 Security Support will run out before even 3.9 (by a few months). Keeping 7.2 in 3.10 will make this way worse.

**Comment by bratkartoffel** [comment]
Currently this PR is waiting on #5902 and #5905 to get merged. My tests so far are successfull, so I think that we can merge this into 3.10 soon.

**Comment by Ikke** [comment]
@bratkartoffel  3.10 will be branched any moment now (when 3.10 is released). At that point, it's not trivial anymore to get it into 3.10.

So I'll look at those PRs today to see if they can get get merged.

**Comment by ncopa** [comment]
I think we may need have `pcre2-dev` in `depends_dev`. I suggest this:

````diff
diff --git a/community/php7/APKBUILD b/community/php7/APKBUILD
index d11de41064..6de1f32eb8 100644
--- a/community/php7/APKBUILD
+++ b/community/php7/APKBUILD
@@ -37,7 +37,7 @@ url="https://secure.php.net"
 arch="all"
 license="PHP-3.01 BSD LGPL-2.0-or-later MIT Zend-2.0"
 depends="$pkgname-common"
-depends_dev="$pkgname=$pkgver-r$pkgrel autoconf pcre-dev"
+depends_dev="$pkgname=$pkgver-r$pkgrel autoconf pcre2-dev"
 # Most dependencies between extensions is auto-discovered (see _extension()).
 _depends_mysqlnd="$pkgname-openssl"
 _depends_pdo_mysql="$pkgname-pdo $pkgname-mysqlnd"
@@ -46,7 +46,7 @@ _depends_phar="$pkgname"
 # load index based on number of dependencies.
 _depends_mysqli="$pkgname-mysqlnd $pkgname-openssl"
 makedepends="
-       autoconf
+       $depends_dev
        apache2-dev
        aspell-dev
        bison
@@ -75,7 +75,6 @@ makedepends="
        libzip-dev
        net-snmp-dev
        openldap-dev
-       pcre2-dev
        postgresql-dev
        re2c
        recode-dev
````

**Comment by bratkartoffel** [comment]
outdated / merged with fda7e51568483590e3ad2ed9ee2f3097d5c3d25e

**Comment by andypost** [comment]
Sadly it was commited before 3.10 and will need a lot of rebuilding dependent packages

**Pull request closed without merging**